provisioning/transport/http/devdoc/http_requirements.md

```diff
@@ -0,0 +1,89 @@
+# azure-iot-provisioning-http Requirements
+
+## Overview
+This module provides HTTP protocol support to communicate with the Azure device provisioning service
+
+```js
+'use strict';
+
+export class Http extends EventEmitter implements DeviceProvisioningTransport {
+  api-version: string;
+  operationPollingIntervalInSeconds: Number
+  constructor(idScope: string, registrationId: string) : void;
+  register(authorization?: string, forceRegistration?: boolean, done(err: Error, result?: Any) => void) : void;
+}
+```
+
+## Public Interface
+
+### constructor(idScope: string, registrationId: string) : void
+The `constructor` creates an Http transport object used to communicate with the Azure device provisioning service
+
+**SRS_NODE_PROVISIONING_HTTP_18_001: [** The `Http` constructor shall accept the following properties:
+- `idScope` - a string specifiying the scope of the provisioning operations,
+- `registrationId` - the registration id for the specific device 
+**]**
+
+
+### register(authorization?: string, forceRegistration?: boolean, callback (err: Error, result?: Any) => void) : void;
+`register` calls into the service to register the given device with the provisioning service.
+
+**SRS_NODE_PROVISIONING_HTTP_18_002: [** `register` shall throw a `ReferenceError` if `callback` is falsy. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_004: [** The registration request shall use Http 1.1. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_005: [** The registration request shall include the current `api-version` as a URL query string value. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_006: [** The registration request shall specify the following in the Http header:
+  Accept: application/json
+  Content-Type: application/json; charset=utf-8 **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_007: [** If an `authorization` string is specifed, the registration request shall include this in the Http header. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_008: [** If `forceRegistration` is specified, the registration request shall include this as a query string value. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_009: [** `register` shall shall POST the registration request to 'https://global.azure-devices-provisioning.net/{idScope}/registrations/{registrationId}' **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_010: [** `register` shall wait for a registration response. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_011: [** If the registration request times out, `register` shall call the `callback` with `TimeoutError` error **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_012: [** If the registration response contains a body, `register` shall deserialize this into a JSON object. **]**
```

Comment:
>JSON [](start = 128, length = 4)

Elio would say that JSON is a notation and by definition a string :) "into an object" is simple enough - and maybe needs a type?

Reply:
Good point.  I don't want to tie this to a type yet.  Even though the schema appears well defined at this point, I don't have a reason  to enforce that schema yet.

---
In reply to: [140060986](https://github.com/Azure/azure-iot-sdk-node/pull/86#discussion_r140060986) [](ancestors = 140060986)

Reply:
"yet" nevertheless it'll make it easier for users to have autocompletion with at least an interface. even if for now you leave it as simple as a basic generic dictionnary, it will give us a clue not to forget it.

---
In reply to: [140107605](https://github.com/Azure/azure-iot-sdk-node/pull/86#discussion_r140107605) [](ancestors = 140107605,140060986)